api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3807,6 +3815,25 @@ private long getAvailableSize(Object data, long expectedReadSize)
     return totalBytesRead;
   }
 
+  private void fillData(ByteArrayOutputStream buffer, Object data, long size) throws IOException {
+    buffer.reset();
+    byte[] buf = new byte[16384]; // 16KiB buffer for optimization
+    long totalBytesRead = 0;
+    while (totalBytesRead < size) {
+      long bytesToRead = size - totalBytesRead;
+      if (bytesToRead > buf.length) bytesToRead = buf.length;
+      int bytesRead = -1;
+      if (data instanceof RandomAccessFile) {
+        bytesRead = ((RandomAccessFile) data).read(buf, 0, (int) bytesToRead);
+      } else if (data instanceof BufferedInputStream) {
+        bytesRead = ((BufferedInputStream) data).read(buf, 0, (int) bytesToRead);
+      }
+      if (bytesRead < 0) throw new IOException("EOF on reading data");
+      buffer.write(buf, 0, bytesRead);
```

Comment:
If the offset is always passed as `0`, will it not keep overwriting the data written in the previous iteration?

Reply:
No